app/javascript/dashboard/components/ChatList.vue

```diff
@@ -350,7 +390,10 @@ export default {
       this.resetAndFetchData();
     },
     activeFolder() {
-      if (!this.hasAppliedFilters) {
+      if (
+        !this.hasAppliedFilters ||
+        (this.isLastSavedFolderOpen === false && this.$route.name !== 'home')
```

Comment:
This change will help when a new filter is applied and saved as a folder and will redirects to the last folder view, after that when we click another saved folder it will now fetch that selected filter when the filter is active.